apps/drec-api/src/pods/file/file.controller.ts

```diff
@@ -0,0 +1,109 @@
+import { FILE_SUPPORTED_MIMETYPES } from '@energyweb/origin-backend-core';
+import {
+  Controller,
+  Get,
+  HttpStatus,
+  NotFoundException,
+  Param,
+  Post,
+  Res,
+  UploadedFiles,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { FileFieldsInterceptor } from '@nestjs/platform-express';
+import {
+  ApiBearerAuth,
+  ApiBody,
+  ApiConsumes,
+  ApiNotFoundResponse,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { Request, Response } from 'express';
+import multer from 'multer';
+
+import { FileDto } from './file.dto';
+import { FileUploadDto } from './file-upload.dto';
+import { FileService } from './file.service';
+import { Roles } from '../user/decorators/roles.decorator';
+import { RolesGuard } from '../../auth/roles-guard';
+import { Role } from '../../utils/eums';
+import { UserDecorator } from '../user/decorators/user.decorator';
+import { OrganizationUserDTO } from '../../auth/dto/org-user.dto';
+
+const maxFilesLimit = parseInt(process.env.FILE_MAX_FILES, 10) || 20;
+const maxFileSize = parseInt(process.env.FILE_MAX_FILE_SIZE, 10) || 10485760;
+
+@ApiTags('file')
+@ApiBearerAuth('access-token')
+@Controller('file')
+@UseGuards(AuthGuard('jwt'), RolesGuard)
+@Roles(Role.Admin, Role.DeviceOwner)
+export class FileController {
+  constructor(private readonly fileService: FileService) {}
+
+  @Post()
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({ type: FileUploadDto })
+  @UseInterceptors(
+    FileFieldsInterceptor([{ name: 'files', maxCount: maxFilesLimit }], {
```

Comment:
Maybe it's me, but AFAIR multer doesn't interrupt file upload until it's completely uploaded, and THEN it breaks the upload. Which is quite serious issue.

Can you verify that?

Reply:
Hmm not sure how I can test this in the code, but I can do a bit of research.

Reply:
Just send giant file and in the network tab see if connection is interrupted or not. If it is it means that everything is loaded by node.

We had this problem on another project, but maybe something changed in the multer lib in the meantime

Reply:
I can only test it with Postman. If the file exceeds the Max Length then it just throws an error that file is too large. So we should be ok.

Reply:
Yeah, sure, but how quickly? It's hard to test locally, because transfer is almost immediate.

Let's just assume that this is not the problem :)

Reply:
It was pretty fast (108ms)